packages/@lwc/engine-core/src/framework/upgradable-element.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined, isFunction } from '@lwc/shared';
+import { LightningElement } from './base-lightning-element';
+import { Renderer } from './renderer';
+
```

Comment:
the problem is that this method here might need to return a regular CE rather than an upgradable element. Having `upgrade` static is not a requirement, but a nice to have.